pkg/types/contract_reader.go

```diff
@@ -112,6 +123,11 @@ type Sequence struct {
 	Data any
 }
 
+type SequenceWithKey struct {
```

Comment:
It looks like this will still be necessary, but can it be an unexported implementation detail now?

Reply:
agree, no longer needs to be public, removed